storage_splunk_hec.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// SplunkHecConfig describes the YAML-provided configuration for a Splunk HEC
+// storage backend.
+type SplunkHecConfig struct {
+	Token                     string `yaml:"token"`
+	Tenant                    string `yaml:"tenant"`
+	Port                      int    `yaml:"port"`
+	Host                      string `yaml:"host"`
+	Source                    string `yaml:"source"`
+	MetricsSourceType         string `yaml:"metrics-source-type"`
+	MetricsIndex              string `yaml:"metrics-index"`
+	EventsSourceType          string `yaml:"events-source-type"`
+	EventsIndex               string `yaml:"events-index"`
+	SkipCertificateValidation bool   `yaml:"skip-cert-validation"`
```

Comment:
splunk cloud ssl certs seem to be invalid for many clients (postman, go http.Client), added this flag to allow bypassing the issue

Reply:
Alternatively, we could allow users to pass a CA cert file location and tell them how to go download that from Splunk.  I guess Splunk uses its own private CA.

Reply:
Splunk is using its own cert for these endpoints, do you have an example how to set the cert?

Reply:
I was looking through the docs.  As far as I can tell:
- They use, by default, a self-signed cert for their endpoint 
- They don't provide this cert to users as far as I can tell. Maybe packaged with a client library?
- They can optionally install a server cert that you provide onto the endpoint.  For this use case (which is probably what we're doing at Okta), we need a way to provide the **CA cert** to crabby so that it can validate the server's cert.

Reply:
full splunk cloud accounts (not trials) have the right certificate. splunk enterprise can be configured with your own certs. I'll leave it as is for now and work on the provide additional ca certs after merge. would that work?